inc/classes/ServiceProvider/class-common-subscribers.php

```diff
@@ -52,7 +52,7 @@ public function register() {
 			->withArgument( $this->getContainer()->get( 'db_optimization' ) )
 			->withArgument( $this->getContainer()->get( 'options' ) );
 		$this->getContainer()->add( 'critical_css_generation', 'WP_Rocket\Optimization\CSS\Critical_CSS_Generation' );
-		$this->getContainer()->add( 'critical_css', 'WP_Rocket\Optimization\CSS\Critical_CSS' )
+		$this->getContainer()->add( 'critical_css', 'WP_Rocket\Engine\Optimization\CSS\CriticalCSS' )
```

Comment:
Shouldn't all of the critical CSS be in the optimization service provider? @Tabrisrp @crystinutzaa

Reply:
It's currently in the common because the CriticalCSS subscriber has code for the admin and for the front at the same time (and some for both, for example for regeneration triggered from the admin bar).

And the Optimization service provider is loaded only when it's _not_ the admin, so it wouldn't work.

It's of course possible to change that if we feel it's necessary. We can either:
- Load the optimization service provider always, but I don't think it's great
- Split the Critical CSS subscriber in multiple parts, loaded in the appropriate place (one for admin, one for front, one for common)

Reply:
Makes sense. @Tabrisrp We want to move both the common and optimization service providers into our new architecture. What do you think makes sense for bundling everything?

Reply:
IMO, the critical CSS subscriber is currently doing too much (the whole class has high complexity, and 13 public methods)

So if we want to tackle that right now, what makes the most sense to me is breaking it into 3 parts.

Once this is done, we can move each ServiceProvider (Admin/Common/Optimization) into the new architecture.

Reply:
Other ways to architecture it are:

1. Have a Critical CSS module that contains all of its files (self-contained) including its own `ServiceProvider`. 
2. Have an `Optimization` module where each submodule has all of its own files (self-contained) and then there's one `ServiceProvider` for the entire `Optimization` module.

Remember, the idea with the new architecture is to have complete modules and sub-modules.

@Tabrisrp What do you think?